### ISSUE TITLE
Rule `letter-case` does not fix `unicodeEscape`

### ISSUE THREAD

**Issue by yvele** [ota-meshi/eslint-plugin-regexp#743]
**Information:**
- ESLint version: 8.57.0
- `eslint-plugin-regexp` version: 2.6.0

**Description**

With this rule elabled:

```js
"regexp/letter-case" : ["error", {
  caseInsensitive : "ignore",
  unicodeEscape : "lowercase",
  hexadecimalEscape : "lowercase",
  controlEscape : "uppercase"
}]
```

The following code:

```js
const REG2 = /\u036F/u;
```

is reported as error:

> error  '\u036F' is not in lowercase               regexp/letter-case

but should be automatically fixed to:

```diff
- const REG2 = /\u036F/u;
+ const REG2 = /\u036f/u;
```

Note that [the rule doc](https://ota-meshi.github.io/eslint-plugin-regexp/rules/letter-case.html) specifies that:

> 🔧 This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

**Comment by yvele** [comment]
Ok I find out that there is a conflict between the `regexp/letter-case` default options: https://ota-meshi.github.io/eslint-plugin-regexp/rules/letter-case.html#options

```js
{
  "regexp/letter-case": ["error", {
    "unicodeEscape": "lowercase"
  }]
}
```

And the recommended `unicorn/escape-case` rule: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/escape-case.md

> Enforces defining escape sequence values with uppercase characters rather than lowercase ones. This promotes readability by making the escaped value more distinguishable from the identifier.

As an example let's see how core-js is doing: https://github.com/zloirock/core-js/blob/f1303f38bd80474f950cadb3e802db1c2618a2c5/tests/eslint/eslint.config.js#L705-L707C21

```js
  // require escape sequences to use uppercase values
  'unicorn/escape-case': ERROR,

  'regexp/letter-case': [ERROR, {
    caseInsensitive: 'lowercase',
    unicodeEscape: 'uppercase',
  }],
```

Maybe we could explain that conflict in the doc? Because `eslint-plugin-unicorn` is super widely used.

**Comment by ota-meshi** [comment]
I welcome pull requests to add notes.
